### PR TITLE
Refactoring front-end [wip]

### PR DESCRIPTION
**Pull request by fabricionaweb**
Applied webpack-rails for frontend sources;

* webpack 2.2rc0
* babel

Created tree-shaking structure for used components (called with `data-component` attr)

### CODE REVIEW

File: app/views/maps/_form_edit.html.erb [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
 
-    <button class="button button-secundary add-sub-sections">Adicionar sub-seções</button>
-  <% end %>
-</div>
+  <%= f.input :department, wrapper: false, include_blank: false,
+    input_html: { class: 'u-mb-small', data: { department_to: true } },
+    collection: departments, label: false %>
+
+  <button type="button" data-save-modal class="button button-primary">
+    Salvar
```
Comment (dmitryrck) [inline app/views/maps/_form_edit.html.erb]: Locale :P

File: app/views/maps/edit.html.erb [hunk 2]
```diff
@@ -1,53 +1,39 @@
...
       </div>
 
-      <%= hidden_field :map, :store_id, value: @map.store_id %>
-      <%= hidden_field :map, :id, value: @map.id %>
-
-      <div class="align-right">
-        <%= link_to 'Publicar', "#", id: 'updateMap',
-            class: 'button button-primary u-right u-mb-small' %>
-      </div>
+      <button data-save-map class="button button-primary u-float-right u-mb-small">Publicar</button>
```
Comment (dmitryrck) [inline app/views/maps/edit.html.erb]: Locale :P

File: app/views/navigation/_menu.html.erb [hunk 3]
```diff
@@ -1,14 +1,10 @@
...
 
-    <button class="button button-text toggle-navigation"
-      data-trigger-close-menu="side-menu">
+    <button class="button button-text toggle-navigation" data-close-navigation-menu>
       <i class="glyph glyph-plus"></i>
-
-      <span class="close-text">
-        fechar
-      </span>
+      <span class="close-text">fechar</span>
```
Comment (dmitryrck) [inline app/views/navigation/_menu.html.erb]: Locale :P

File: app/views/sessions/new.html.erb [hunk 4]
```diff
@@ -1,29 +1,37 @@
...
         <% end %>
 
-        <div class="col-xs-12 u-mt-medium">
+        <div class="u-mt-medium">
           <%= f.submit 'Entrar', class: 'button button-primary u-fill' %>
         </div>
 
-        <div class="col-xs-12 u-mt-medium align-center">
+        <div class="u-mt-medium align-center">
           <%= link_to 'Esqueci minha senha', new_user_password_path %>
```
Comment (dmitryrck) [inline app/views/sessions/new.html.erb]: Eita, cuidado com a polícia do locale :P :policewoman:

File: webpack/components/editor-map.js [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+  }
+
+  onSaveMapSuccess (response) {
+    window.location.reload()
+  }
+
+  onSaveMapWarning (error) {
+    console.warn(error)
+    window.location.reload()
+    // Loading.hide()
```
Comment (dmitryrck) [inline webpack/components/editor-map.js]: COMMENTS DETECTED :P

File: webpack/styles/field.css [hunk 6]
```diff
@@ -0,0 +1,11 @@
+/* fix input label z-index */
```
Comment (dmitryrck) [inline webpack/styles/field.css]: COMMENT DETECTED :P

Comment (fabricionaweb) [inline webpack/styles/field.css]: Esse é para explicar o que se passa mesmo ahhaha

Comment (dmitryrck) [inline webpack/styles/field.css]: :)

**Pull request merged**